duktape/src/main/java/com/squareup/duktape/Duktape.java

```diff
@@ -95,6 +97,33 @@ public String evaluate(String script) {
   }
 
   /**
+   * Creates a proxy to a global JavaScript object called {@code name} that implements {@code type}.
+   * {@code type} defines the interface implemented in JavaScript that will be accessible to Java.
+   * {@code type} must be an interface that does not extend any other interfaces, and cannot define
+   * any overloaded methods.
+   * <p>
+   * Methods of the interface may return {@code void} or any of the following supported argument
+   * types: {@code boolean}, {@link Boolean}, {@code int}, {@link Integer}, {@code double},
+   * {@link Double}, {@link String}.
+   * </p>
+   */
+  public <T> T proxy(final String name, Class<T> type) {
+    if (!type.isInterface()) {
+      throw new UnsupportedOperationException("Only interfaces can be proxied. Received: " + type);
+    }
+    if (type.getInterfaces().length > 0) {
+      throw new UnsupportedOperationException(type + " must not extend other interfaces");
+    }
+    return (T) Proxy.newProxyInstance(type.getClassLoader(), new Class<?>[]{ type },
+        new InvocationHandler() {
+      @Override
+      public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
```

Comment:
Can you check if `method`’s declaring class is `type`? All interfaces inherit the methods of Object.class and when that happens your best bet is to do say `return method.invoke(InvocationHandler.this, args)` instead of calling out to JavaScript.

(and then if you also want, you can override `toString()` in this invocation handler to return something like `"proxy(value, FooInterface.class)"`